### PR TITLE
Bump go version to 1.19.2 and update hashes

### PR DESCRIPTION
**Pull request by HamzaAnis**
Updated the max Golang version to latest 1.19.2

### CODE REVIEW

**Review by Jimexist — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module github.com/apache/thrift
 
-go 1.18
+go 1.19.2
```
Comment (fishy) [inline go.mod:3]: this is wrong:
1. this cannot include point releases
2. this is the minimal version of go we support, and we don't want to drop support for go1.18 yet, per our support policy at https://github.com/apache/thrift/blob/master/lib/go/README.md#suppored-go-releases

Comment (fishy) [reply go.mod:3]: @Jimexist this is also reported by the github action workflow for go: https://github.com/apache/thrift/actions/runs/3214860554/jobs/5257904020

Comment (HamzaAnis) [reply go.mod:3]: @fishy I believe we can revert this. I created this PR considering two minor go releases but you guys are considering two major go releases.

Comment (fishy) [reply go.mod:3]: 1. in semver `1` is major and `18`/`19` is minor
2. https://github.com/apache/thrift/pull/2702

**Pull request merged**